contracts/src/v0.8/dev/ocr2dr/OCR2DRRegistry.sol

```diff
@@ -0,0 +1,783 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.4;
+
+import "../../interfaces/LinkTokenInterface.sol";
+import "../../interfaces/AggregatorV3Interface.sol";
+import "../interfaces/OCR2DRRegistryInterface.sol";
+import "../interfaces/OCR2DRBillableInterface.sol";
+import "../interfaces/OCR2DRClientInterface.sol";
+import "../../interfaces/TypeAndVersionInterface.sol";
+import "../../interfaces/ERC677ReceiverInterface.sol";
+import "../../ConfirmedOwner.sol";
+
+contract OCR2DRRegistry is ConfirmedOwner, TypeAndVersionInterface, OCR2DRRegistryInterface, ERC677ReceiverInterface {
+  LinkTokenInterface public immutable LINK;
+  AggregatorV3Interface public immutable LINK_ETH_FEED;
+
+  // We need to maintain a list of consuming addresses.
+  // This bound ensures we are able to loop over them as needed.
+  // Should a user require more consumers, they can use multiple subscriptions.
+  uint16 public constant MAX_CONSUMERS = 100;
+  // 5k is plenty for an EXTCODESIZE call (2600) + warm CALL (100)
+  // and some arithmetic operations.
+  uint256 private constant GAS_FOR_CALL_EXACT_CHECK = 5_000;
+  // Maximum number of oracles DON can support
+  // Needs to match OCR2Abstract.sol
+  uint256 internal constant MAX_NUM_ORACLES = 31;
```

Comment:
I don't like we duplicate this value in different places. Clearly we should either inject this in constructor, or do something else.

Reply:
Removed it instead because signers is unused.

Reply:
@pinebit I re-added this so that the DON payment could be split across NOPs.

I don't see a way to inject this in the constructor, because it is used within the interface for `concludeBilling`.
I suppose I could remove that function signature from the interface. Then initialize it within the Registry.
Then have the Oracle contracts pull from the Registry's value.
But even then I'm not sure it will work.

It also doesn't make sense to make it a dynamic array, because then it would need to be held in storage. It can change on every report, because not every node may sign.

Let me know if you can think of anyway around it.